addons/purchase_stock/models/res_partner.py

```diff
@@ -17,24 +18,31 @@ class ResPartner(models.Model):
 
     @api.depends('purchase_line_ids')
     def _compute_on_time_rate(self):
-        order_lines = self.env['purchase.order.line'].search([
+        order_line_ids = self.env['purchase.order.line']._search([
             ('partner_id', 'in', self.ids),
             ('date_order', '>', fields.Date.today() - timedelta(365)),
             ('qty_received', '!=', 0),
-            ('order_id.state', 'in', ['done', 'purchase'])
-        ]).filtered(lambda l: l.product_id.sudo().product_tmpl_id.type != 'service')
+            ('order_id.state', 'in', ['done', 'purchase'])])
+
```

Comment:
maybe add a limit : 1000.

Reply:
> Idea : limit the compute of last 10k po.line (with a ir.config.parameter)

I'm not trying to change the current behavior and the value seen in the on_tim_rate, simply computing it in chunks